lib/dnsimple/client/certificates.rb

```diff
@@ -108,6 +108,167 @@ def certificate_private_key(account_id, domain_id, certificate_id, options = {})
+      #   certificate.name            # => "www"
+      #   certificate.common_name     # => "www.example.com"
+      #   certificate.alternate_names # => []
+      #   certificate.auto_renew      # => true
+      def letsencrypt_purchase(account_id, domain_id, attributes, options = {})
+        Extra.validate_mandatory_attributes(attributes, [:contact_id])
+        response = client.post(Client.versioned("/%s/domains/%s/certificates/letsencrypt" % [account_id, domain_id]), attributes, options)
+
+        Dnsimple::Response.new(response, Struct::Certificate.new(response["data"]))
+      end
+
+      # Issue a pending Let's Encrypt certificate order.
+      #
+      # Note that the issuance process is async. A successful response means the issuance
+      # request has been successfully acknowledged and queued for processing.
+      #
+      # @see https://developer.dnsimple.com/v2/domains/certificates/#letsencrypt-issue
+      #
+      # @param  account_id [Integer] the account ID
+      # @param  domain_id [#to_s] the domain ID or domain name
+      # @param  certificate_id [Integer] the certificate ID returned by the purchase method
+      # @param  options [Hash]
+      #
+      # @return [Dnsimple::Response<Dnsimple::Struct::Certificate>]
+      #
+      # @raise  [Dnsimple::NotFoundError]
+      # @raise  [Dnsimple::RequestError]
+      #
+      # @example Basic usage
+      #   reponse     = client.certificates.letsencrypt_issue(1010, "example.com", 100)
+      #   certificate = response.data
+      #
+      #   certificate.state # => "requesting"
+      def letsencrypt_issue(account_id, domain_id, certificate_id, options = {})
+        response = client.post(Client.versioned("/%s/domains/%s/certificates/letsencrypt/%s/issue" % [account_id, domain_id, certificate_id]), options)
+
+        Dnsimple::Response.new(response, Struct::Certificate.new(response["data"]))
+      end
+
+      # Purchase a Let's Encrypt certificate renewal.
+      #
+      # @see https://developer.dnsimple.com/v2/domains/certificates/#letsencrypt-purchase-renewal
+      #
+      # @param  account_id [Integer] the account ID
+      # @param  domain_id [#to_s] the domain ID or domain name
+      # @param  certificate_id [Integer] the certificate ID
+      # @param  attributes [Hash]
+      # @option attributes [TrueClass,FalseClass] :auto_renew enable certificate auto renew (optional)
+      # @param  options [Hash]
+      #
+      # @return [Dnsimple::Response<Dnsimple::Struct::CertificateRenewal>]
+      #
+      # @raise  [Dnsimple::NotFoundError]
+      # @raise  [Dnsimple::RequestError]
+      #
+      # @example Basic usage
+      #   reponse             = client.certificates.letsencrypt_purchase(1010, "example.com", 200)
+      #   certificate_renewal = response.data
+      #
+      #   certificate_renewal.id                 # => 999
+      #   certificate_renewal.old_certificate_id # => 200
+      #   certificate_renewal.new_certificate_id # => 200
```

Comment:
This is confusing that it is the same ID as the old certificate id. Is there a case where they are actually the same certificate?

Reply:
It's a typo, that should be `300`. Sorry.